util/aggregation.py

```diff
@@ -435,13 +443,23 @@ def aggregate(async_job_dir, request_id=None):
         os.makedirs(final_dir)
 
     try:
-        aggregate_status(local_dir, final_dir, request_id=request_id)
-        aggregate_json(local_dir, final_dir, request_id=request_id)
-        aggregate_csv(local_dir, final_dir, request_id=request_id)
-        aggregate_netcdf(local_dir, final_dir, request_id=request_id)
-        aggregate_provenance(local_dir, final_dir, request_id=request_id)
-        aggregate_annotations(local_dir, final_dir, request_id=request_id)
-        generate_ncml(final_dir, final_dir, request_id=request_id)
+        # check for empty local_dir
+        if not os.listdir(local_dir):
```

Comment:
I'm assuming that it's more likely local_dir will contain files; if that's so I recommend changing the "if" to check for a populated local_dir, etc.

Reply:
indeed, if the user attempted to request data that doesn't exist, this error more than likely should be a 400 series error, but if the data is supposed to exist and there should be files here, then a 500 series error would be correct.  also, since this is being written to a file, is the http status code still relevant here, or is it just following how existing status messages are being written to the file?

Reply:
Yes, it is more likely local_dir will contain files -- will change if statement as suggested.

Yes, calling aggregation on an empty directory is more of a symptom, although at this point it indicates something bad happened (i.e. an internal error). I will change the error message to exclude the internal error indication.

If a user requested data that does not exist, either A) EDEX should return an appropriate error message and aggregation does not get called (e.g. if stream metadata table indicates no data) or B) status.txt and/or failure.json files should be written to the intermediate results directory (the one that gets aggregated) and therefore the directory would not be empty. The various stream engine routes have try-catch blocks around their calculating logic so they should write status files even in error cases. All this means that the presence of an empty intermediate results directory indicates an unexpected failure to write status files.

The HTTP status code is just for conforming to how existing status messages are being written to the file.